src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        /// <see cref="SetCalibration(PgaSensitivity, float, float)"/> <seealso cref="SetCalibration(ushort)"/>
+        /// </remarks>
+        public float PowerLsb { get; set; }
+
+        /// <summary>
+        /// Construct an Ina219 device using an I2cDevice
+        /// </summary>
+        /// <remarks>
+        /// This binding does not dispose the passed in I2cDevice.
+        /// </remarks>
+        /// <param name="i2cDevice">The I2cDevice initialized to communicate with the Ina219.</param>
+        public Ina219(I2cDevice i2cDevice)
+        {
+            if (i2cDevice == null)
+            {
+                throw new System.ArgumentNullException(nameof(i2cDevice));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Construct an Ina219 device using an I2cConnectionSettings.
+        /// </summary>
+        /// <remarks>
+        /// This binding creates an I2cDevice ufor communication with the Ina219. The I2cDevice is disposed when then Ina219 is disposed.
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropiate settings to communicate with the Ina219.</param>
+        public Ina219(I2cConnectionSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new System.ArgumentNullException(nameof(settings));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+            _calibrationValue = 0;
+
+            _disposeI2cDevice = true;
+            _i2cDevice = I2cDevice.Create(settings);
+        }
+
+        /// <summary>
+        /// Reset the Ina219 to default values;
+        /// </summary>
+        public void Reset()
+        {
+            ushort regValue;
+
+            // Reset the device by sending a value to the configuration register with the reset but set.
+            WriteRegister(Register.Configuration, (ushort)ConfigurationFlags.Rst);
+
+            // Read back the values from the configuration register which should now be default.
+            regValue = ReadRegister(Register.Configuration, 1000);
+
+            PowerLsb = PowerLsb = 1F;
+            _calibrationValue = 0;
+
+            // Get the current values for the Bus and Shunt ADC settings.
+            _busAdcResSamp = (AdcResolutionOrSamples)((regValue & (ushort)ConfigurationFlags.BadcMask) >> 4);
+            _shuntAdcResSamp = (AdcResolutionOrSamples)(regValue & (ushort)ConfigurationFlags.SadcMask);
+        }
+
+        /// <summary>
+        /// Set the Ina219 operating mode.
+        /// </summary>
+        /// <param name="mode">The operating mode to switch to.</param>
+        public void SetOperatingMode(OperatingMode mode)
+        {
+            ushort regValue = ReadRegister(Register.Configuration);
+
+            regValue &= (ushort)ConfigurationFlags.ModeMask ^ 0xFFFF;
```

Comment:
double checking: did you mean to flip bits here or did you mean `& 0xFFFF`

Reply:
And with the inverse of the mask.

Reply:
so this will flip instead :smile: i.e. `0b1010 ^ 0b1111` will become `0b0101` instead of zeros

Reply:
seems functionally correct, but why not just use ~ if you want to flip all bits?

Reply:
@maloo 
You are right of course.